src/components/PopupDelete.js

```diff
@@ -0,0 +1,24 @@
+import Popup from "./Popup.js";
+
+export default class PopupDelete extends Popup {
+  constructor(popupSelector, handleDeleteConfirm) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._handleDeleteConfirm = handleDeleteConfirm;
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  open() {
+    this._submitButton.textContent = "Save";
```

Comment:
Needs correction: please don't change the button text here. You should create a separate method for rendering loading and start the loading right before the api request and end the loading after in `finally` block